### ISSUE TITLE
There should be a possibility to declare map key as a sequence

### ISSUE THREAD

**Issue by jarrodek** [raml-org/raml-spec#245]
Consider following example:

```yaml
/files:
  get:
    responses:
      200:
        body:
          application/json:
            type: files.File
  post:
    responses:
      200:
        body:
          application/json:
            type: files.File
```
Both `GET` and `POST` request will result with the same response. It's redundant to declare it separately. It could be defined as follows:
```yaml
/files:
  [get, post]:
    responses:
      200:
        body:
          application/json:
            type: files.File
```
Yaml specification allows using sequences as a map key so the change would be aligning RAML with Yaml.

Looking forward for any feedback.

**Comment by sichvoge** [comment]
+1, that is definitely very helpful and should be possible since RAML is based on YAML. In fact, RAML supports the DRY principle with many features such as traits and resourceTypes. Therefore, that would definitely play very well with our methodologies. IMHO!

**Comment by usarid** [comment]
But we can already achieve this via
```raml
traits:
  file-responding:
    responses:
      200:
        body:
          application/json:
            type: files.File    
/files:
  get:
    is: file-responding
  post:
    is: file-responding
```
and this is a much more flexible structure as you can start differentiating between them too -- as is inevitable, I think.

**Comment by sichvoge** [comment]
That you could do indeed - its the making it much more flexible using traits. Although you still have room for improvements in your example combining both approaches.

```yaml
traits:
  file-responding:
    responses:
      200:
        body:
          application/json:
            type: files.File    
/files:
  [get, post]:
    is: file-responding
```

Use cases here could be where the most common HTTP all have the same response structure (maybe rare, but that can happen. I think about CRUD operations that always returns an file object.

```yaml
traits:
  file-responding:
    responses:
      200:
        body:
          application/json:
            type: files.File    
/files:
  [get, post, put, delete]:
    is: file-responding
```

**Comment by jarrodek** [comment]
You are right @usarid But isn't it redundant again? Like @sichvoge suggested in the example above it could be only 2 line definition instead of at least for (depending on an API).
